sdk/metric/controller/push/push.go

```diff
@@ -153,14 +156,36 @@ func (c *Controller) Stop() {
 	c.ch = nil
 	c.wg.Wait()
 	c.ticker.Stop()
+	c.ticker = nil
 
 	c.tick()
+
+	close(c.configLoaderCh)
+}
+
+func (c *Controller) RestartTicker(period time.Duration) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	// Stop the existing ticker
+	// Make a new ticker with the new sampling period
+	c.ticker.Stop()
+	c.period = period
+	c.ticker = c.clock.Ticker(period)
+
+	// Let the controller know to check the new ticker
+	c.ch <- true
 }
 
-func (c *Controller) run(ch chan struct{}) {
+func (c *Controller) run(ch chan bool) {
 	for {
 		select {
-		case <-ch:
+		// If signal receives 'true', break to check the new ticker
```

Comment:
Unregister the watcher here.

It is okay for the notifier to outlive the controller, so we don't need to bother with closing the notifier here.

Reply:
I unregister the watcher in the controller's `stop` method, if that's okay